# Importar librerías


In [2]:
import pandas as pd
import numpy as np


# Importar base de datos desde Google drive

In [3]:
from google.colab import drive
drive.mount("/content/drive")
archivo= "/content/drive/MyDrive/Delay_2023_12.csv"
df=pd.read_csv(archivo, sep= ",", header=0)
df

Mounted at /content/drive


<ipython-input-3-71aa1284e785>:4: DtypeWarning: Columns (69,76,77,84) have mixed types. Specify dtype option on import or set low_memory=False.
  df=pd.read_csv(archivo, sep= ",", header=0)


Year  Quarter  Month  DayofMonth  DayOfWeek  FlightDate  \
0       2023        4     12          25          1  2023-12-25   
1       2023        4     12          25          1  2023-12-25   
2       2023        4     12          25          1  2023-12-25   
3       2023        4     12          25          1  2023-12-25   
4       2023        4     12          20          3  2023-12-20   
...      ...      ...    ...         ...        ...         ...   
570389  2023        4     12          31          7  2023-12-31   
570390  2023        4     12          31          7  2023-12-31   
570391  2023        4     12          31          7  2023-12-31   
570392  2023        4     12          31          7  2023-12-31   
570393  2023        4     12          31          7  2023-12-31   

       Reporting_Airline  DOT_ID_Reporting_Airline  \
0                     OO                     20304   
1                     OO                     20304   
2                     OO                     20304   
3                     OO                     20304   
4                     OO                     20304   
...                  ...                       ...   
570389                B6                     20409   
570390                B6                     20409   
570391                B6                     20409   
570392                B6                     20409   
570393                B6                     20409   

       IATA_CODE_Reporting_Airline Tail_Number  ...  Div4TailNum  Div5Airport  \
0                               OO      N611SK  ...          NaN          NaN   
1                               OO      N762SK  ...          NaN          NaN   
2                               OO      N608SK  ...          NaN          NaN   
3                               OO      N760SK  ...          NaN          NaN   
4                               OO      N797SK  ...          NaN          NaN   
...                            ...         ...  ...          ...          ...   
570389                          B6      N903JB  ...          NaN          NaN   
570390                          B6      N535JB  ...          NaN          NaN   
570391                          B6      N354JB  ...          NaN          NaN   
570392                          B6      N768JB  ...          NaN          NaN   
570393                          B6      N547JB  ...          NaN          NaN   

        Div5AirportID  Div5AirportSeqID Div5WheelsOn Div5TotalGTime  \
0                 NaN               NaN          NaN            NaN   
1                 NaN               NaN          NaN            NaN   
2                 NaN               NaN          NaN            NaN   
3                 NaN               NaN          NaN            NaN   
4                 NaN               NaN          NaN            NaN   
...               ...               ...          ...            ...   
570389            NaN               NaN          NaN            NaN   
570390            NaN               NaN          NaN            NaN   
570391            NaN               NaN          NaN            NaN   
570392            NaN               NaN          NaN            NaN   
570393            NaN               NaN          NaN            NaN   

       Div5LongestGTime  Div5WheelsOff Div5TailNum  Unnamed: 109  
0                   NaN            NaN         NaN           NaN  
1                   NaN            NaN         NaN           NaN  
2                   NaN            NaN         NaN           NaN  
3                   NaN            NaN         NaN           NaN  
4                   NaN            NaN         NaN           NaN  
...                 ...            ...         ...           ...  
570389              NaN            NaN         NaN           NaN  
570390              NaN            NaN         NaN           NaN  
570391              NaN            NaN         NaN           NaN  
570392              NaN            NaN         NaN        

# Entendiendo los datos

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 570394 entries, 0 to 570393
Columns: 110 entries, Year to Unnamed: 109
dtypes: float64(70), int64(21), object(19)
memory usage: 478.7+ MB


In [38]:
df.columns

Index(['Year', 'Quarter', 'Month', 'DayofMonth', 'DayOfWeek', 'FlightDate',
       'Reporting_Airline', 'DOT_ID_Reporting_Airline',
       'IATA_CODE_Reporting_Airline', 'Tail_Number',
       ...
       'Div4TailNum', 'Div5Airport', 'Div5AirportID', 'Div5AirportSeqID',
       'Div5WheelsOn', 'Div5TotalGTime', 'Div5LongestGTime', 'Div5WheelsOff',
       'Div5TailNum', 'Unnamed: 109'],
      dtype='object', length=110)

# Realizamos una copia de los datos

In [4]:
delay= df.copy()
delay

Year  Quarter  Month  DayofMonth  DayOfWeek  FlightDate  \
0       2023        4     12          25          1  2023-12-25   
1       2023        4     12          25          1  2023-12-25   
2       2023        4     12          25          1  2023-12-25   
3       2023        4     12          25          1  2023-12-25   
4       2023        4     12          20          3  2023-12-20   
...      ...      ...    ...         ...        ...         ...   
570389  2023        4     12          31          7  2023-12-31   
570390  2023        4     12          31          7  2023-12-31   
570391  2023        4     12          31          7  2023-12-31   
570392  2023        4     12          31          7  2023-12-31   
570393  2023        4     12          31          7  2023-12-31   

       Reporting_Airline  DOT_ID_Reporting_Airline  \
0                     OO                     20304   
1                     OO                     20304   
2                     OO                     20304   
3                     OO                     20304   
4                     OO                     20304   
...                  ...                       ...   
570389                B6                     20409   
570390                B6                     20409   
570391                B6                     20409   
570392                B6                     20409   
570393                B6                     20409   

       IATA_CODE_Reporting_Airline Tail_Number  ...  Div4TailNum  Div5Airport  \
0                               OO      N611SK  ...          NaN          NaN   
1                               OO      N762SK  ...          NaN          NaN   
2                               OO      N608SK  ...          NaN          NaN   
3                               OO      N760SK  ...          NaN          NaN   
4                               OO      N797SK  ...          NaN          NaN   
...                            ...         ...  ...          ...          ...   
570389                          B6      N903JB  ...          NaN          NaN   
570390                          B6      N535JB  ...          NaN          NaN   
570391                          B6      N354JB  ...          NaN          NaN   
570392                          B6      N768JB  ...          NaN          NaN   
570393                          B6      N547JB  ...          NaN          NaN   

        Div5AirportID  Div5AirportSeqID Div5WheelsOn Div5TotalGTime  \
0                 NaN               NaN          NaN            NaN   
1                 NaN               NaN          NaN            NaN   
2                 NaN               NaN          NaN            NaN   
3                 NaN               NaN          NaN            NaN   
4                 NaN               NaN          NaN            NaN   
...               ...               ...          ...            ...   
570389            NaN               NaN          NaN            NaN   
570390            NaN               NaN          NaN            NaN   
570391            NaN               NaN          NaN            NaN   
570392            NaN               NaN          NaN            NaN   
570393            NaN               NaN          NaN            NaN   

       Div5LongestGTime  Div5WheelsOff Div5TailNum  Unnamed: 109  
0                   NaN            NaN         NaN           NaN  
1                   NaN            NaN         NaN           NaN  
2                   NaN            NaN         NaN           NaN  
3                   NaN            NaN         NaN           NaN  
4                   NaN            NaN         NaN           NaN  
...                 ...            ...         ...           ...  
570389              NaN            NaN         NaN           NaN  
570390              NaN            NaN         NaN           NaN  
570391              NaN            NaN         NaN           NaN  
570392              NaN            NaN         NaN        

# Calculamos información de los datos, observaciones y variables

In [ ]:
print(f"la base de datos tiene {delay.shape[0]} observaciones y {delay.shape[1]} variables")

la base de datos tiene 570394 observaciones y 110 variables


# Calculamos datos duplicados

In [116]:
print(f"la base de datos tiene {delay.duplicated().sum()} observaciones duplicadas")

la base de datos tiene 0 observaciones duplicadas


# Calculamos datos nulos

In [25]:
delay.isnull().sum()

Year                     0
Quarter                  0
Month                    0
DayofMonth               0
DayOfWeek                0
                     ...  
Div5TotalGTime      570394
Div5LongestGTime    570394
Div5WheelsOff       570394
Div5TailNum         570394
Unnamed: 109        570394
Length: 110, dtype: int64

In [26]:
delay.isnull().sum().sum()

30373485

In [29]:
#Datos nulos a binarios
delay.isnull()*1

Year  Quarter  Month  DayofMonth  DayOfWeek  FlightDate  \
0          0        0      0           0          0           0   
1          0        0      0           0          0           0   
2          0        0      0           0          0           0   
3          0        0      0           0          0           0   
4          0        0      0           0          0           0   
...      ...      ...    ...         ...        ...         ...   
570389     0        0      0           0          0           0   
570390     0        0      0           0          0           0   
570391     0        0      0           0          0           0   
570392     0        0      0           0          0           0   
570393     0        0      0           0          0           0   

        Reporting_Airline  DOT_ID_Reporting_Airline  \
0                       0                         0   
1                       0                         0   
2                       0                         0   
3                       0                         0   
4                       0                         0   
...                   ...                       ...   
570389                  0                         0   
570390                  0                         0   
570391                  0                         0   
570392                  0                         0   
570393                  0                         0   

        IATA_CODE_Reporting_Airline  Tail_Number  ...  Div4TailNum  \
0                                 0            0  ...            1   
1                                 0            0  ...            1   
2                                 0            0  ...            1   
3                                 0            0  ...            1   
4                                 0            0  ...            1   
...                             ...          ...  ...          ...   
570389                            0            0  ...            1   
570390                            0            0  ...            1   
570391                            0            0  ...            1   
570392                            0            0  ...            1   
570393                            0            0  ...            1   

        Div5Airport  Div5AirportID  Div5AirportSeqID  Div5WheelsOn  \
0                 1              1                 1             1   
1                 1              1                 1             1   
2                 1              1                 1             1   
3                 1              1                 1             1   
4                 1              1                 1             1   
...             ...            ...               ...           ...   
570389            1              1                 1             1   
570390            1              1                 1             1   
570391            1              1                 1             1   
570392            1              1                 1             1   
570393            1              1                 1             1   

        Div5TotalGTime  Div5LongestGTime  Div5WheelsOff  Div5TailNum  \
0                    1                 1              1            1   
1                    1                 1              1            1   
2                    1                 1              1            1   
3                    1                 1              1            1   
4                    1                 1              1            1   
...                ...               ...            ...          ...   
570389               1                 1              1            1   
570390               1                 1              1            1   
570391               1                 1              1            1   
570392               1                 1              1            1   
570393               1                 1              1            1

In [30]:
#Columnas con datos nulos
delay.isnull().any()

Year                False
Quarter             False
Month               False
DayofMonth          False
DayOfWeek           False
                    ...  
Div5TotalGTime       True
Div5LongestGTime     True
Div5WheelsOff        True
Div5TailNum          True
Unnamed: 109         True
Length: 110, dtype: bool

# Datos nulos por variable

In [32]:
delay.isnull().sum()

Year                     0
Quarter                  0
Month                    0
DayofMonth               0
DayOfWeek                0
                     ...  
Div5TotalGTime      570394
Div5LongestGTime    570394
Div5WheelsOff       570394
Div5TailNum         570394
Unnamed: 109        570394
Length: 110, dtype: int64


# Tratamiento de los datos nulos

In [35]:
# Eliminamos filas si todos los valores en la fila son faltantes
delay2= delay.dropna(how="all")
delay2

Year  Quarter  Month  DayofMonth  DayOfWeek  FlightDate  \
0       2023        4     12          25          1  2023-12-25   
1       2023        4     12          25          1  2023-12-25   
2       2023        4     12          25          1  2023-12-25   
3       2023        4     12          25          1  2023-12-25   
4       2023        4     12          20          3  2023-12-20   
...      ...      ...    ...         ...        ...         ...   
570389  2023        4     12          31          7  2023-12-31   
570390  2023        4     12          31          7  2023-12-31   
570391  2023        4     12          31          7  2023-12-31   
570392  2023        4     12          31          7  2023-12-31   
570393  2023        4     12          31          7  2023-12-31   

       Reporting_Airline  DOT_ID_Reporting_Airline  \
0                     OO                     20304   
1                     OO                     20304   
2                     OO                     20304   
3                     OO                     20304   
4                     OO                     20304   
...                  ...                       ...   
570389                B6                     20409   
570390                B6                     20409   
570391                B6                     20409   
570392                B6                     20409   
570393                B6                     20409   

       IATA_CODE_Reporting_Airline Tail_Number  ...  Div4TailNum  Div5Airport  \
0                               OO      N611SK  ...          NaN          NaN   
1                               OO      N762SK  ...          NaN          NaN   
2                               OO      N608SK  ...          NaN          NaN   
3                               OO      N760SK  ...          NaN          NaN   
4                               OO      N797SK  ...          NaN          NaN   
...                            ...         ...  ...          ...          ...   
570389                          B6      N903JB  ...          NaN          NaN   
570390                          B6      N535JB  ...          NaN          NaN   
570391                          B6      N354JB  ...          NaN          NaN   
570392                          B6      N768JB  ...          NaN          NaN   
570393                          B6      N547JB  ...          NaN          NaN   

        Div5AirportID  Div5AirportSeqID Div5WheelsOn Div5TotalGTime  \
0                 NaN               NaN          NaN            NaN   
1                 NaN               NaN          NaN            NaN   
2                 NaN               NaN          NaN            NaN   
3                 NaN               NaN          NaN            NaN   
4                 NaN               NaN          NaN            NaN   
...               ...               ...          ...            ...   
570389            NaN               NaN          NaN            NaN   
570390            NaN               NaN          NaN            NaN   
570391            NaN               NaN          NaN            NaN   
570392            NaN               NaN          NaN            NaN   
570393            NaN               NaN          NaN            NaN   

       Div5LongestGTime  Div5WheelsOff Div5TailNum  Unnamed: 109  
0                   NaN            NaN         NaN           NaN  
1                   NaN            NaN         NaN           NaN  
2                   NaN            NaN         NaN           NaN  
3                   NaN            NaN         NaN           NaN  
4                   NaN            NaN         NaN           NaN  
...                 ...            ...         ...           ...  
570389              NaN            NaN         NaN           NaN  
570390              NaN            NaN         NaN           NaN  
570391              NaN            NaN         NaN           NaN  
570392              NaN            NaN         NaN        

In [31]:
# Eliminamos columnas con al menos un valor NaN
delay2= delay.dropna(axis=1)
delay2

Year  Quarter  Month  DayofMonth  DayOfWeek  FlightDate  \
0       2023        4     12          25          1  2023-12-25   
1       2023        4     12          25          1  2023-12-25   
2       2023        4     12          25          1  2023-12-25   
3       2023        4     12          25          1  2023-12-25   
4       2023        4     12          20          3  2023-12-20   
...      ...      ...    ...         ...        ...         ...   
570389  2023        4     12          31          7  2023-12-31   
570390  2023        4     12          31          7  2023-12-31   
570391  2023        4     12          31          7  2023-12-31   
570392  2023        4     12          31          7  2023-12-31   
570393  2023        4     12          31          7  2023-12-31   

       Reporting_Airline  DOT_ID_Reporting_Airline  \
0                     OO                     20304   
1                     OO                     20304   
2                     OO                     20304   
3                     OO                     20304   
4                     OO                     20304   
...                  ...                       ...   
570389                B6                     20409   
570390                B6                     20409   
570391                B6                     20409   
570392                B6                     20409   
570393                B6                     20409   

       IATA_CODE_Reporting_Airline  Flight_Number_Reporting_Airline  ...  \
0                               OO                             3173  ...   
1                               OO                             3178  ...   
2                               OO                             3205  ...   
3                               OO                             3232  ...   
4                               OO                             5772  ...   
...                            ...                              ...  ...   
570389                          B6                             2130  ...   
570390                          B6                             2133  ...   
570391                          B6                             2160  ...   
570392                          B6                             2161  ...   
570393                          B6                             2162  ...   

        DepTimeBlk  CRSArrTime  ArrTimeBlk Cancelled Diverted CRSElapsedTime  \
0        1200-1259        1325   1300-1359       0.0      0.0           65.0   
1        1200-1259        1356   1300-1359       0.0      0.0           86.0   
2        0600-0659        1019   1000-1059       0.0      0.0          159.0   
3        0600-0659         738   0700-0759       0.0      0.0          158.0   
4        1100-1159        1235   1200-1259       0.0      0.0           64.0   
...            ...         ...         ...       ...      ...            ...   
570389   0800-0859        1140   1100-1159       0.0      0.0          256.0   
570390   1800-1859        2202   2200-2259       0.0      0.0          175.0   
570391   1300-1359        1419   1400-1459       0.0      0.0           74.0   
570392   1600-1659        1946   1900-1959       0.0      0.0          181.0   
570393   1000-1059        1406   1400-1459       0.0      0.0          189.0   

        Flights Distance  DistanceGroup  DivAirportLandings  
0           1.0    134.0              1                   0  
1           1.0    295.0              2                   0  
2           1.0   1022.0              5                   0  
3           1.0    870.0              4                   0  
4           1.0    125.0              1                   0  
...         ...      ...            ...                 ...  
570389      1.0   1598.0              7                   0  
570390      1.0   1189.0              5                   0  
570391      1.0    280.0              2                   0  
570392      1.0   1133.0              5                   0 

# Entendiendo los datos Data Frame delay2

In [37]:
delay2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 570394 entries, 0 to 570393
Data columns (total 39 columns):
 #   Column                           Non-Null Count   Dtype  
---  ------                           --------------   -----  
 0   Year                             570394 non-null  int64  
 1   Quarter                          570394 non-null  int64  
 2   Month                            570394 non-null  int64  
 3   DayofMonth                       570394 non-null  int64  
 4   DayOfWeek                        570394 non-null  int64  
 5   FlightDate                       570394 non-null  object 
 6   Reporting_Airline                570394 non-null  object 
 7   DOT_ID_Reporting_Airline         570394 non-null  int64  
 8   IATA_CODE_Reporting_Airline      570394 non-null  object 
 9   Flight_Number_Reporting_Airline  570394 non-null  int64  
 10  OriginAirportID                  570394 non-null  int64  
 11  OriginAirportSeqID               570394 non-null  int64  
 12  Or

In [39]:
#Un análisis de las variables numericas
delay2.describe()

Year   Quarter     Month     DayofMonth      DayOfWeek  \
count  570394.0  570394.0  570394.0  570394.000000  570394.000000   
mean     2023.0       4.0      12.0      16.013931       4.173098   
std         0.0       0.0       0.0       8.913793       1.987917   
min      2023.0       4.0      12.0       1.000000       1.000000   
25%      2023.0       4.0      12.0       8.000000       2.000000   
50%      2023.0       4.0      12.0      16.000000       4.000000   
75%      2023.0       4.0      12.0      24.000000       6.000000   
max      2023.0       4.0      12.0      31.000000       7.000000   

       DOT_ID_Reporting_Airline  Flight_Number_Reporting_Airline  \
count             570394.000000                    570394.000000   
mean               19933.133680                      2289.156590   
std                  377.058368                      1558.489996   
min                19393.000000                         1.000000   
25%                19790.000000                      1046.000000   
50%                19805.000000                      2043.000000   
75%                20363.000000                      3343.000000   
max                20452.000000                      8819.000000   

       OriginAirportID  OriginAirportSeqID  OriginCityMarketID  ...  \
count    570394.000000        5.703940e+05       570394.000000  ...   
mean      12659.526249        1.265957e+06        31745.336713  ...   
std        1530.980316        1.530978e+05         1316.863343  ...   
min       10135.000000        1.013506e+06        30070.000000  ...   
25%       11292.000000        1.129202e+06        30647.000000  ...   
50%       12889.000000        1.288904e+06        31454.000000  ...   
75%       14057.000000        1.405702e+06        32467.000000  ...   
max       16869.000000        1.686902e+06        35991.000000  ...   

             DestWac     CRSDepTime     CRSArrTime      Cancelled  \
count  570394.000000  570394.000000  570394.000000  570394.000000   
mean       54.891053    1328.260783    1490.279198       0.004050   
std        26.668717     496.326834     525.829108       0.063509   
min         1.000000       1.000000       1.000000       0.000000   
25%        34.000000     905.000000    1102.000000       0.000000   
50%        51.000000    1322.000000    1518.000000       0.000000   
75%        82.000000    1740.000000    1928.000000       0.000000   
max        93.000000    2359.000000    2359.000000       1.000000   

            Diverted  CRSElapsedTime   Flights       Distance  DistanceGroup  \
count  570394.000000   570394.000000  570394.0  570394.000000  570394.000000   
mean        0.002004      149.815878       1.0     844.389178       3.850721   
std         0.044720       73.648672       0.0     596.917602       2.334784   
min         0.000000     -272.000000       1.0      31.000000       1.000000   
25%         0.000000       95.000000       1.0     405.000000       2.000000   
50%         0.000000      134.000000       1.0     700.000000       3.000000   
75%         0.000000      181.000000       1.0    1076.000000       5.000000   
max         1.000000     1003.000000       1.0    5095.000000      11.000000   

       DivAirportLandings  
count       570394.000000  
mean             0.003115  
std              0.108473  
min              0.000000  
25%              0.000000  
50%              0.000000  
75%              0.000000  
max              9.000000  

[8 rows x 26 columns]

In [ ]:
delay2.columns

Index(['Year', 'Quarter', 'Month', 'DayofMonth', 'DayOfWeek', 'FlightDate',
       'Reporting_Airline', 'DOT_ID_Reporting_Airline',
       'IATA_CODE_Reporting_Airline', 'Flight_Number_Reporting_Airline',
       'OriginAirportID', 'OriginAirportSeqID', 'OriginCityMarketID', 'Origin',
       'OriginCityName', 'OriginState', 'OriginStateFips', 'OriginStateName',
       'OriginWac', 'DestAirportID', 'DestAirportSeqID', 'DestCityMarketID',
       'Dest', 'DestCityName', 'DestState', 'DestStateFips', 'DestStateName',
       'DestWac', 'CRSDepTime', 'DepTimeBlk', 'CRSArrTime', 'ArrTimeBlk',
       'Cancelled', 'Diverted', 'CRSElapsedTime', 'Flights', 'Distance',
       'DistanceGroup', 'DivAirportLandings'],
      dtype='object')

# Analítica

In [117]:
#Identificando cuantos años se registran en la base de datos
delay2['Year'].unique()

array([2023])

In [53]:
#Identificando cuantos meses se registran en la base de datos
delay2['Month'].unique()

array([12])

In [ ]:
#Encontramos las aerolineas  que hacen parte de este reporte (las aerolíneas estan con su codigo y no por su nombre)
delay2['Reporting_Airline'].unique()

array(['OO', 'UA', 'B6', 'DL', 'F9', 'G4', 'HA', 'MQ', 'NK', 'OH', 'WN',
       'YX', '9E', 'AA', 'AS'], dtype=object)

##Origen y destino de los vuelos

In [40]:
#Estados desde donde se originan los vuelos
delay2['OriginStateName'].unique()

array(['Texas', 'Louisiana', 'Arizona', 'Kansas', 'Colorado', 'Alabama',
       'Wyoming', 'Nevada', 'Georgia', 'Illinois', 'Michigan',
       'California', 'New Mexico', 'Minnesota', 'Utah', 'Nebraska',
       'Oregon', 'Oklahoma', 'Wisconsin', 'Indiana', 'Washington',
       'Idaho', 'North Dakota', 'New York', 'South Dakota',
       'Pennsylvania', 'Montana', 'Iowa', 'Tennessee', 'Ohio', 'Virginia',
       'North Carolina', 'Arkansas', 'Kentucky', 'South Carolina',
       'Missouri', 'Massachusetts', 'Florida', 'Maine', 'Mississippi',
       'Connecticut', 'West Virginia', 'Maryland', 'New Jersey',
       'Vermont', 'Puerto Rico', 'U.S. Virgin Islands', 'Alaska',
       'Hawaii', 'U.S. Pacific Trust Territories and Possessions',
       'Rhode Island', 'New Hampshire'], dtype=object)

In [122]:
#Estados con mas vuelos destino
delay2['DestStateName'].value_counts()

DestStateName
California                                        61806
Texas                                             60779
Florida                                           57050
Georgia                                           29904
New York                                          29343
Illinois                                          28538
Colorado                                          28163
North Carolina                                    23581
Virginia                                          18572
Arizona                                           18426
Nevada                                            17543
Washington                                        14953
Michigan                                          12473
Tennessee                                         11831
Pennsylvania                                      11533
Hawaii                                            11108
New Jersey                                        11049
Massachusetts                     

In [41]:
#Ciudades destino de los vuelos
delay2['DestCityName'].unique()

array(['Dallas/Fort Worth, TX', 'Phoenix, AZ', 'Denver, CO', 'Eagle, CO',
       'Atlanta, GA', 'Salt Lake City, UT', 'Los Angeles, CA',
       'Montgomery, AL', 'Detroit, MI', 'Monterey, CA',
       'Palm Springs, CA', 'San Francisco, CA', 'Shreveport, LA',
       'Elmira/Corning, NY', 'Minneapolis, MN', 'St. George, UT',
       'Aspen, CO', 'Bend/Redmond, OR', 'Grand Island, NE',
       'Longview, TX', 'Houston, TX', 'San Luis Obispo, CA',
       'Harlingen/San Benito, TX', 'Brunswick, GA', 'Evansville, IN',
       'Medford, OR', 'Chicago, IL', 'Pittsburgh, PA', 'Washington, DC',
       'Duluth, MN', 'Omaha, NE', 'Oakland, CA', 'Seattle, WA',
       'Tucson, AZ', 'New York, NY', 'Nashville, TN', 'Spokane, WA',
       'Idaho Falls, ID', 'Bakersfield, CA', 'Boise, ID', 'Reno, NV',
       'Bozeman, MT', 'Fresno, CA', 'Great Falls, MT', 'Portland, OR',
       'Roanoke, VA', 'Knoxville, TN', 'Pasco/Kennewick/Richland, WA',
       'Missoula, MT', 'Grand Rapids, MI', 'Sioux Falls, SD',
    

In [120]:
#Ciudades con mas vuelos destino
delay2['DestCityName'].value_counts()

DestCityName
Atlanta, GA                      28052
Chicago, IL                      27498
Denver, CO                       25057
Dallas/Fort Worth, TX            23680
New York, NY                     22827
                                 ...  
St. Cloud, MN                       10
Pago Pago, TT                        9
Adak Island, AK                      9
Santa Maria, CA                      8
Newport News/Williamsburg, VA        2
Name: count, Length: 328, dtype: int64

In [ ]:
# Reporte del arrivo de vuelos por aerolínea
dfDep_ArrDelay = delay2[['Reporting_Airline','ArrTimeBlk']]
dfDep_ArrDelay

Reporting_Airline ArrTimeBlk
0                     OO  1300-1359
1                     OO  1300-1359
2                     OO  1000-1059
3                     OO  0700-0759
4                     OO  1200-1259
...                  ...        ...
570389                B6  1100-1159
570390                B6  2200-2259
570391                B6  1400-1459
570392                B6  1900-1959
570393                B6  1400-1459

[570394 rows x 2 columns]

# Estadistica por columnas

In [123]:
from ipywidgets import interact

In [124]:
delay2.columns

Index(['Year', 'Quarter', 'Month', 'DayofMonth', 'DayOfWeek', 'FlightDate',
       'Reporting_Airline', 'DOT_ID_Reporting_Airline',
       'IATA_CODE_Reporting_Airline', 'Flight_Number_Reporting_Airline',
       'OriginAirportID', 'OriginAirportSeqID', 'OriginCityMarketID', 'Origin',
       'OriginCityName', 'OriginState', 'OriginStateFips', 'OriginStateName',
       'OriginWac', 'DestAirportID', 'DestAirportSeqID', 'DestCityMarketID',
       'Dest', 'DestCityName', 'DestState', 'DestStateFips', 'DestStateName',
       'DestWac', 'CRSDepTime', 'DepTimeBlk', 'CRSArrTime', 'ArrTimeBlk',
       'Cancelled', 'Diverted', 'CRSElapsedTime', 'Flights', 'Distance',
       'DistanceGroup', 'DivAirportLandings'],
      dtype='object')

## Tablero Interactivo

Este tablero interactivo nos muestra los datos del Data Frame por columna y nos permite navegar facilmente entre cada variable

In [125]:
@interact(columna = ['Year', 'Quarter', 'Month', 'DayofMonth', 'DayOfWeek', 'FlightDate',
       'Reporting_Airline', 'DOT_ID_Reporting_Airline',
       'IATA_CODE_Reporting_Airline', 'Flight_Number_Reporting_Airline',
       'OriginAirportID', 'OriginAirportSeqID', 'OriginCityMarketID', 'Origin',
       'OriginCityName', 'OriginState', 'OriginStateFips', 'OriginStateName',
       'OriginWac', 'DestAirportID', 'DestAirportSeqID', 'DestCityMarketID',
       'Dest', 'DestCityName', 'DestState', 'DestStateFips', 'DestStateName',
       'DestWac', 'CRSDepTime', 'DepTimeBlk', 'CRSArrTime', 'ArrTimeBlk',
       'Cancelled', 'Diverted', 'CRSElapsedTime', 'Flights', 'Distance',
       'DistanceGroup', 'DivAirportLandings'])

def vistaInteractiva(columna):
  dataAux = delay2[columna]
  print(dataAux)
  print('DataFrame con', delay2.ndim)

interactive(children=(Dropdown(description='columna', options=('Year', 'Quarter', 'Month', 'DayofMonth', 'DayO…

# Analisis de los vuelos

##Conteo de Vuelos

In [84]:
# Conteo de los dias del mes con mayor numero de vuelos
delay.DayofMonth.value_counts()

DayofMonth
22    19840
21    19752
15    19558
14    19484
7     19432
8     19415
29    19283
26    19207
11    19206
10    19195
28    19185
27    19147
4     19108
1     19095
20    19071
3     19009
23    18744
18    18676
17    18619
30    18504
19    18122
13    17976
12    17786
6     17697
5     17495
31    16752
16    16476
25    16326
2     16151
9     16122
24    15961
Name: count, dtype: int64

In [85]:
# Conteo de los dias de la semana con mayor numero de vuelos
delay.DayOfWeek.value_counts()

DayOfWeek
5    97191
7    89536
6    85997
4    77853
3    73891
1    73316
2    72610
Name: count, dtype: int64

##Desvios

In [136]:
#Total de vuelos desviados
delay2['Diverted'].value_counts()

Diverted
0.0    569251
1.0      1143
Name: count, dtype: int64

Segun el calculo anterior la base de datos tiene 1143 vuelos desviados

In [142]:
#Conteo Vuelos desviados según la aerolinea
delay2[['Reporting_Airline','Diverted']].value_counts()

Reporting_Airline  Diverted
WN                 0.0         127595
DL                 0.0          80927
AA                 0.0          75284
UA                 0.0          59759
OO                 0.0          55738
NK                 0.0          22559
YX                 0.0          21580
B6                 0.0          21315
MQ                 0.0          20379
AS                 0.0          19339
9E                 0.0          16858
OH                 0.0          16100
F9                 0.0          15676
G4                 0.0           9445
HA                 0.0           6697
OO                 1.0            258
WN                 1.0            178
AA                 1.0            124
DL                 1.0             95
UA                 1.0             85
B6                 1.0             83
YX                 1.0             61
AS                 1.0             61
OH                 1.0             41
MQ                 1.0             41
G4                 1.0

In [143]:
#Conteo Estados donde se presentan mayores vuelos desviados
delay2[['OriginStateName','Diverted']].value_counts()

OriginStateName  Diverted
California       0.0         61670
Texas            0.0         60704
Florida          0.0         56851
Georgia          0.0         29826
New York         0.0         29318
                             ...  
Montana          1.0             2
Kansas           1.0             2
Wyoming          1.0             2
South Dakota     1.0             1
Mississippi      1.0             1
Name: count, Length: 100, dtype: int64

In [144]:
#Conteo Ciudades donde se presentan mayores vuelos desviados
delay2[['OriginCityName','Diverted']].value_counts()

OriginCityName                Diverted
Atlanta, GA                   0.0         27976
Chicago, IL                   0.0         27436
Denver, CO                    0.0         24984
Dallas/Fort Worth, TX         0.0         23646
New York, NY                  0.0         22802
                                          ...  
Liberal, KS                   1.0             1
Traverse City, MI             1.0             1
Lihue, HI                     1.0             1
Rock Springs, WY              1.0             1
Pasco/Kennewick/Richland, WA  1.0             1
Name: count, Length: 465, dtype: int64

In [145]:
# conteo de los dias del mes con mas desvios
delay2[['DayofMonth','Diverted']].value_counts()

DayofMonth  Diverted
22          0.0         19788
21          0.0         19720
15          0.0         19538
14          0.0         19458
7           0.0         19418
                        ...  
13          1.0            16
28          1.0            16
19          1.0            14
7           1.0            14
31          1.0            13
Name: count, Length: 62, dtype: int64

In [146]:
# conteo de los dias de la semana con mas desvios
delay2[['DayOfWeek','Diverted']].value_counts()

DayOfWeek  Diverted
5          0.0         97034
7          0.0         89290
6          0.0         85771
4          0.0         77765
3          0.0         73771
1          0.0         73139
2          0.0         72481
7          1.0           246
6          1.0           226
1          1.0           177
5          1.0           157
2          1.0           129
3          1.0           120
4          1.0            88
Name: count, dtype: int64

##Cancelados

In [137]:
#Total de vuelos cancelados
delay2['Cancelled'].value_counts()

Cancelled
0.0    568084
1.0      2310
Name: count, dtype: int64

Segun el calculo anterior la base de datos tiene 2310 vuelos cancelados

In [147]:
#Conteo Vuelos cancelados según la aerolinea
delay2[['Reporting_Airline','Cancelled']].value_counts()

Reporting_Airline  Cancelled
WN                 0.0          126967
DL                 0.0           81000
AA                 0.0           75372
UA                 0.0           59767
OO                 0.0           55575
NK                 0.0           22485
YX                 0.0           21510
B6                 0.0           21315
MQ                 0.0           20378
AS                 0.0           19272
9E                 0.0           16732
OH                 0.0           16069
F9                 0.0           15622
G4                 0.0            9418
HA                 0.0            6602
WN                 1.0             806
OO                 1.0             421
9E                 1.0             156
YX                 1.0             131
AS                 1.0             128
HA                 1.0             100
NK                 1.0              98
B6                 1.0              83
UA                 1.0              77
F9                 1.0             

In [148]:
#Conteo Estados donde se presentan mayores vuelos cancelados
delay2[['OriginStateName','Cancelled']].value_counts()

OriginStateName      Cancelled
California           0.0          61595
Texas                0.0          60708
Florida              0.0          56807
Georgia              0.0          29848
New York             0.0          29168
                                  ...  
Oklahoma             1.0              5
Kansas               1.0              5
U.S. Virgin Islands  1.0              2
Arkansas             1.0              2
Mississippi          1.0              1
Name: count, Length: 102, dtype: int64

In [149]:
#Conteo Ciudades donde se presentan mayores vuelos cancelados
delay2[['OriginCityName','Cancelled']].value_counts()

OriginCityName                 Cancelled
Atlanta, GA                    0.0          28003
Chicago, IL                    0.0          27272
Denver, CO                     0.0          24959
Dallas/Fort Worth, TX          0.0          23694
New York, NY                   0.0          22699
                                            ...  
Little Rock, AR                1.0              1
Kodiak, AK                     1.0              1
Nome, AK                       1.0              1
Peoria, IL                     1.0              1
Sun Valley/Hailey/Ketchum, ID  1.0              1
Name: count, Length: 533, dtype: int64

In [150]:
#Conteo de los dias del mes con mas vuelos cancelados
delay2[['DayofMonth','Cancelled']].value_counts()

DayofMonth  Cancelled
22          0.0          19778
21          0.0          19698
15          0.0          19533
14          0.0          19450
7           0.0          19402
                         ...  
31          1.0             24
30          1.0             23
19          1.0             22
9           1.0             21
5           1.0             21
Name: count, Length: 62, dtype: int64

In [151]:
#Conteo de los dias de la semana  con mas vuelos cancelados
delay2[['DayOfWeek','Cancelled']].value_counts()

DayOfWeek  Cancelled
5          0.0          96994
7          0.0          88998
6          0.0          85695
4          0.0          77700
3          0.0          73665
1          0.0          72670
2          0.0          72362
1          1.0            646
7          1.0            538
6          1.0            302
2          1.0            248
3          1.0            226
5          1.0            197
4          1.0            153
Name: count, dtype: int64

In [152]:
#Conteo Lapsos de tiempo de las aerolineas
delay2[['Reporting_Airline','CRSElapsedTime']].value_counts()

Reporting_Airline  CRSElapsedTime
WN                 85.0              7998
                   90.0              7585
                   80.0              6743
                   75.0              5793
                   70.0              5602
                                     ... 
G4                 288.0                1
HA                 30.0                 1
                   300.0                1
                   515.0                1
F9                 331.0                1
Name: count, Length: 3608, dtype: int64

# Aerolineas

In [88]:
#Aerolineas por mayor numero de vuelos
delay.Reporting_Airline.value_counts()

Reporting_Airline
WN    127773
DL     81022
AA     75408
UA     59844
OO     55996
NK     22583
YX     21641
B6     21398
MQ     20420
AS     19400
9E     16888
OH     16141
F9     15698
G4      9480
HA      6702
Name: count, dtype: int64

In [154]:
# retraso y arrivos de vuelos,  'DepTimeBlk' (retrasos) y 'ArrTimeBlk' (arrivos)
dfDep_ArrDelay = delay2[['DepTimeBlk','ArrTimeBlk']]
dfDep_ArrDelay

DepTimeBlk ArrTimeBlk
0       1200-1259  1300-1359
1       1200-1259  1300-1359
2       0600-0659  1000-1059
3       0600-0659  0700-0759
4       1100-1159  1200-1259
...           ...        ...
570389  0800-0859  1100-1159
570390  1800-1859  2200-2259
570391  1300-1359  1400-1459
570392  1600-1659  1900-1959
570393  1000-1059  1400-1459

[570394 rows x 2 columns]

In [89]:
#Grupos de distancias recorridas
delay2.DistanceGroup.value_counts()

DistanceGroup
2     125978
3     112470
4      97976
5      70855
1      62659
6      26681
7      25808
10     14099
8      12486
11     12083
9       9299
Name: count, dtype: int64

In [79]:
#Grupos de distancia por aerolineas
delay2[['Reporting_Airline','DistanceGroup']]

Reporting_Airline  DistanceGroup
0                     OO              1
1                     OO              2
2                     OO              5
3                     OO              4
4                     OO              1
...                  ...            ...
570389                B6              7
570390                B6              5
570391                B6              2
570392                B6              5
570393                B6              5

[570394 rows x 2 columns]

In [32]:
# retraso y arrivos de vuelos,  'DepTimeBlk' (retrasos) y 'ArrTimeBlk' (arrivos)
dfDep_ArrDelay = delay2[['DepTimeBlk','ArrTimeBlk']]
dfDep_ArrDelay

DepTimeBlk ArrTimeBlk
0       1200-1259  1300-1359
1       1200-1259  1300-1359
2       0600-0659  1000-1059
3       0600-0659  0700-0759
4       1100-1159  1200-1259
...           ...        ...
570389  0800-0859  1100-1159
570390  1800-1859  2200-2259
570391  1300-1359  1400-1459
570392  1600-1659  1900-1959
570393  1000-1059  1400-1459

[570394 rows x 2 columns]

# Tratamiento de las variables


## Variable Retrasos (DepTimeBlk)

In [34]:
##Separamos la columna "DepTimeBlk" en dos columnas "scheduled departure", "true output"
dfDep_ArrDelay[["scheduled departure",
                "true output",
                ]]= dfDep_ArrDelay["DepTimeBlk"].str.split("-", expand=True)
dfDep_ArrDelay

<ipython-input-34-39a28c0a22a8>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dfDep_ArrDelay[["scheduled departure",
<ipython-input-34-39a28c0a22a8>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dfDep_ArrDelay[["scheduled departure",


DepTimeBlk ArrTimeBlk scheduled departure true output
0       1200-1259  1300-1359                1200        1259
1       1200-1259  1300-1359                1200        1259
2       0600-0659  1000-1059                0600        0659
3       0600-0659  0700-0759                0600        0659
4       1100-1159  1200-1259                1100        1159
...           ...        ...                 ...         ...
570389  0800-0859  1100-1159                0800        0859
570390  1800-1859  2200-2259                1800        1859
570391  1300-1359  1400-1459                1300        1359
570392  1600-1659  1900-1959                1600        1659
570393  1000-1059  1400-1459                1000        1059

[570394 rows x 4 columns]

##Variable arrivos (ArrTimeBlk)

In [35]:
#Separamos la columna "ArrTimeBlk" en dos columnas "scheduled arrival", "true arrival"
dfDep_ArrDelay[["scheduled arrival",
                "true arrival",
                ]]= dfDep_ArrDelay["ArrTimeBlk"].str.split("-", expand=True)
dfDep_ArrDelay

<ipython-input-35-0f063321e362>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dfDep_ArrDelay[["scheduled arrival",
<ipython-input-35-0f063321e362>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dfDep_ArrDelay[["scheduled arrival",


DepTimeBlk ArrTimeBlk scheduled departure true output  \
0       1200-1259  1300-1359                1200        1259   
1       1200-1259  1300-1359                1200        1259   
2       0600-0659  1000-1059                0600        0659   
3       0600-0659  0700-0759                0600        0659   
4       1100-1159  1200-1259                1100        1159   
...           ...        ...                 ...         ...   
570389  0800-0859  1100-1159                0800        0859   
570390  1800-1859  2200-2259                1800        1859   
570391  1300-1359  1400-1459                1300        1359   
570392  1600-1659  1900-1959                1600        1659   
570393  1000-1059  1400-1459                1000        1059   

       scheduled arrival true arrival  
0                   1300         1359  
1                   1300         1359  
2                   1000         1059  
3                   0700         0759  
4                   1200         1259  
...                  ...          ...  
570389              1100         1159  
570390              2200         2259  
570391              1400         1459  
570392              1900         1959  
570393              1400         1459  

[570394 rows x 6 columns]

# Tratamiento nuevas variables

##Extracción de variables

In [40]:
#Extraemos las columnas con las variables objeto de estudio
df3= dfDep_ArrDelay[['scheduled departure','true output']]
df3

scheduled departure true output
0                     1200        1259
1                     1200        1259
2                     0600        0659
3                     0600        0659
4                     1100        1159
...                    ...         ...
570389                0800        0859
570390                1800        1859
570391                1300        1359
570392                1600        1659
570393                1000        1059

[570394 rows x 2 columns]

In [42]:
df3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 570394 entries, 0 to 570393
Data columns (total 3 columns):
 #   Column               Non-Null Count   Dtype 
---  ------               --------------   ----- 
 0   scheduled departure  570394 non-null  object
 1   true output          570394 non-null  object
 2   Retraso Salidas      570394 non-null  object
dtypes: object(3)
memory usage: 13.1+ MB


##Transformaciones

In [53]:
#Transformamos las varibales de object a int64
 df3["scheduled departure2"]=df3["scheduled departure"].astype("int64")
df3

<ipython-input-53-84607095da40>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df3["scheduled departure2"]=df3["scheduled departure"].astype("int64")


scheduled departure true output Retraso Salidas  scheduled departure2
0                     1200        1259        12001259                  1200
1                     1200        1259        12001259                  1200
2                     0600        0659        06000659                   600
3                     0600        0659        06000659                   600
4                     1100        1159        11001159                  1100
...                    ...         ...             ...                   ...
570389                0800        0859        08000859                   800
570390                1800        1859        18001859                  1800
570391                1300        1359        13001359                  1300
570392                1600        1659        16001659                  1600
570393                1000        1059        10001059                  1000

[570394 rows x 4 columns]

In [54]:
#Transformamos las varibales de object a int64
df3["true output2"]=df3["true output"].astype("int64")
df3

<ipython-input-54-f0b928c61d8d>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df3["true output2"]=df3["true output"].astype("int64")


scheduled departure true output Retraso Salidas  scheduled departure2  \
0                     1200        1259        12001259                  1200   
1                     1200        1259        12001259                  1200   
2                     0600        0659        06000659                   600   
3                     0600        0659        06000659                   600   
4                     1100        1159        11001159                  1100   
...                    ...         ...             ...                   ...   
570389                0800        0859        08000859                   800   
570390                1800        1859        18001859                  1800   
570391                1300        1359        13001359                  1300   
570392                1600        1659        16001659                  1600   
570393                1000        1059        10001059                  1000   

        true output2  
0               1259  
1               1259  
2                659  
3                659  
4               1159  
...              ...  
570389           859  
570390          1859  
570391          1359  
570392          1659  
570393          1059  

[570394 rows x 5 columns]

# Hallazgos

In [56]:
#obtenemos el retraso total en las salidas de cada vuelo restando las variables 'scheduled departure2' - 'true output2', en la columna "Retraso salidas"
df3 ['Retraso Salidas']= df3['scheduled departure2'] - df3['true output2']
df3

<ipython-input-56-4f19f28e239c>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df3 ['Retraso Salidas']= df3['scheduled departure2'] - df3['true output2'] # nueva columna retraso total sumando las dos columnas anteriores


scheduled departure true output  Retraso Salidas  scheduled departure2  \
0                     1200        1259              -59                  1200   
1                     1200        1259              -59                  1200   
2                     0600        0659              -59                   600   
3                     0600        0659              -59                   600   
4                     1100        1159              -59                  1100   
...                    ...         ...              ...                   ...   
570389                0800        0859              -59                   800   
570390                1800        1859              -59                  1800   
570391                1300        1359              -59                  1300   
570392                1600        1659              -59                  1600   
570393                1000        1059              -59                  1000   

        true output2  
0               1259  
1               1259  
2                659  
3                659  
4               1159  
...              ...  
570389           859  
570390          1859  
570391          1359  
570392          1659  
570393          1059  

[570394 rows x 5 columns]

In [57]:
#Extraemos las columnas con las variables objeto de estudio
dfDep_ArrDelay[["scheduled arrival",
                "true arrival",
                ]]= dfDep_ArrDelay["ArrTimeBlk"].str.split("-", expand=True)
dfDep_ArrDelay

<ipython-input-57-0f063321e362>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dfDep_ArrDelay[["scheduled arrival",


DepTimeBlk ArrTimeBlk scheduled departure true output  \
0       1200-1259  1300-1359                1200        1259   
1       1200-1259  1300-1359                1200        1259   
2       0600-0659  1000-1059                0600        0659   
3       0600-0659  0700-0759                0600        0659   
4       1100-1159  1200-1259                1100        1159   
...           ...        ...                 ...         ...   
570389  0800-0859  1100-1159                0800        0859   
570390  1800-1859  2200-2259                1800        1859   
570391  1300-1359  1400-1459                1300        1359   
570392  1600-1659  1900-1959                1600        1659   
570393  1000-1059  1400-1459                1000        1059   

       scheduled arrival true arrival  
0                   1300         1359  
1                   1300         1359  
2                   1000         1059  
3                   0700         0759  
4                   1200         1259  
...                  ...          ...  
570389              1100         1159  
570390              2200         2259  
570391              1400         1459  
570392              1900         1959  
570393              1400         1459  

[570394 rows x 6 columns]

In [58]:
dfDep_ArrDelay.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 570394 entries, 0 to 570393
Data columns (total 6 columns):
 #   Column               Non-Null Count   Dtype 
---  ------               --------------   ----- 
 0   DepTimeBlk           570394 non-null  object
 1   ArrTimeBlk           570394 non-null  object
 2   scheduled departure  570394 non-null  object
 3   true output          570394 non-null  object
 4   scheduled arrival    570394 non-null  object
 5   true arrival         570394 non-null  object
dtypes: object(6)
memory usage: 26.1+ MB


In [59]:
#Transformamos las varibales de object a int64
df3["scheduled arrival2"]=df3["scheduled departure"].astype("int64")
df3

scheduled departure true output  Retraso Salidas  scheduled departure2  \
0                     1200        1259              -59                  1200   
1                     1200        1259              -59                  1200   
2                     0600        0659              -59                   600   
3                     0600        0659              -59                   600   
4                     1100        1159              -59                  1100   
...                    ...         ...              ...                   ...   
570389                0800        0859              -59                   800   
570390                1800        1859              -59                  1800   
570391                1300        1359              -59                  1300   
570392                1600        1659              -59                  1600   
570393                1000        1059              -59                  1000   

        true output2  scheduled arrival2  
0               1259                1200  
1               1259                1200  
2                659                 600  
3                659                 600  
4               1159                1100  
...              ...                 ...  
570389           859                 800  
570390          1859                1800  
570391          1359                1300  
570392          1659                1600  
570393          1059                1000  

[570394 rows x 6 columns]

In [63]:
#Transformamos las varibales de object a int64
df3["true arrival2"]=df3["true arrival"].astype("int64")
df3

scheduled departure true output  Retraso Salidas  scheduled departure2  \
0                     1200        1259              -59                  1200   
1                     1200        1259              -59                  1200   
2                     0600        0659              -59                   600   
3                     0600        0659              -59                   600   
4                     1100        1159              -59                  1100   
...                    ...         ...              ...                   ...   
570389                0800        0859              -59                   800   
570390                1800        1859              -59                  1800   
570391                1300        1359              -59                  1300   
570392                1600        1659              -59                  1600   
570393                1000        1059              -59                  1000   

        true output2  scheduled arrival2  true arrival  true arrival2  
0               1259                1200          1259           1259  
1               1259                1200          1259           1259  
2                659                 600           659            659  
3                659                 600           659            659  
4               1159                1100          1159           1159  
...              ...                 ...           ...            ...  
570389           859                 800           859            859  
570390          1859                1800          1859           1859  
570391          1359                1300          1359           1359  
570392          1659                1600          1659           1659  
570393          1059                1000          1059           1059  

[570394 rows x 8 columns]

In [64]:
#obtenemos el retraso total en las llegadas de cada vuelo restando las variables 'scheduled arrival2' - 'true arrival2', en la columna "Retraso LLegadas"
df3 ['Retraso LLegadas']= df3['scheduled arrival2'] - df3['true arrival2']
df3

scheduled departure true output  Retraso Salidas  scheduled departure2  \
0                     1200        1259              -59                  1200   
1                     1200        1259              -59                  1200   
2                     0600        0659              -59                   600   
3                     0600        0659              -59                   600   
4                     1100        1159              -59                  1100   
...                    ...         ...              ...                   ...   
570389                0800        0859              -59                   800   
570390                1800        1859              -59                  1800   
570391                1300        1359              -59                  1300   
570392                1600        1659              -59                  1600   
570393                1000        1059              -59                  1000   

        true output2  scheduled arrival2  true arrival  true arrival2  \
0               1259                1200          1259           1259   
1               1259                1200          1259           1259   
2                659                 600           659            659   
3                659                 600           659            659   
4               1159                1100          1159           1159   
...              ...                 ...           ...            ...   
570389           859                 800           859            859   
570390          1859                1800          1859           1859   
570391          1359                1300          1359           1359   
570392          1659                1600          1659           1659   
570393          1059                1000          1059           1059   

        Retraso LLegadas  
0                    -59  
1                    -59  
2                    -59  
3                    -59  
4                    -59  
...                  ...  
570389               -59  
570390               -59  
570391               -59  
570392               -59  
570393               -59  

[570394 rows x 9 columns]

In [71]:
#Obtenemos entonces el retraso total sumando las columnas 'Retraso Salidas' + "Retraso LLegadas"
df3 ['Retraso Total']= df3['Retraso Salidas'] + df3["Retraso LLegadas"]
df3

scheduled departure true output  Retraso Salidas  scheduled departure2  \
0                     1200        1259              -59                  1200   
1                     1200        1259              -59                  1200   
2                     0600        0659              -59                   600   
3                     0600        0659              -59                   600   
4                     1100        1159              -59                  1100   
...                    ...         ...              ...                   ...   
570389                0800        0859              -59                   800   
570390                1800        1859              -59                  1800   
570391                1300        1359              -59                  1300   
570392                1600        1659              -59                  1600   
570393                1000        1059              -59                  1000   

        true output2  scheduled arrival2  true arrival  true arrival2  \
0               1259                1200          1259           1259   
1               1259                1200          1259           1259   
2                659                 600           659            659   
3                659                 600           659            659   
4               1159                1100          1159           1159   
...              ...                 ...           ...            ...   
570389           859                 800           859            859   
570390          1859                1800          1859           1859   
570391          1359                1300          1359           1359   
570392          1659                1600          1659           1659   
570393          1059                1000          1059           1059   

        Retraso LLegadas  Retraso Total  
0                    -59           -118  
1                    -59           -118  
2                    -59           -118  
3                    -59           -118  
4                    -59           -118  
...                  ...            ...  
570389               -59           -118  
570390               -59           -118  
570391               -59           -118  
570392               -59           -118  
570393               -59           -118  

[570394 rows x 10 columns]

# Respuestas

In [99]:
#Retrasos iguales a 0
"Retraso Total"==0

False

In [107]:
retrasos = df3[df3['Retraso Total']<=-15]
N,D = retrasos.shape
print('La cantidad de vuelos que se retrasaron 15 minutos o menos  son:', N)

La cantidad de vuelos que se retrasaron 15 minutos  son: 570394


In [108]:
retrasos = df3[df3['Retraso Total']<=-30]
N,D = retrasos.shape
print('La cantidad de vuelos que se retrasaron  media hora o menos son:', N)

La cantidad de vuelos que se retrasaron  media hora o mas son: 570394


In [111]:
retrasos = df3[df3['Retraso Total']<=-60]
N,D = retrasos.shape
print('La cantidad de vuelos que se retrasaron una hora o menos son:', N)

La cantidad de vuelos que se retrasaron una hora o mas son: 570394


In [112]:
retrasos = df3[df3['Retraso Total']<=-120]
N,D = retrasos.shape
print('La cantidad de vuelos que se retrasaron 2 hora2 o menos son:', N)

La cantidad de vuelos que se retrasaron una hora o mas son: 20075
